### PR TITLE
fix: The lazy-loaded feature can influence the cms mapping for components not covered by it

### LINKED ISSUE #11287: The lazy-loaded feature can influence the cms mapping for components not covered by it
If lazy loaded feature contains Cms mapping definition for a component but feature configuration is not listing this component as a part of this feature, the lazy loading of the feature should not affect cms mapping of this component.

### PR DESCRIPTION
**Pull request by dunqan**
Closes #11287

### CODE REVIEW

File: projects/storefrontlib/src/cms-structure/services/cms-components.service.ts [hunk 1]
```diff
@@ -58,7 +72,8 @@ export class CmsComponentsService {
 
       for (const componentType of componentTypes) {
         if (!this.mappings[componentType]) {
-          const staticConfig = this.config.cmsComponents[componentType];
+          const staticConfig = (this.staticCmsConfig ??
+            this.config.cmsComponents)[componentType];
```
Comment (marlass) [inline projects/storefrontlib/src/cms-structure/services/cms-components.service.ts]: Strict typing error here

File: projects/storefrontlib/src/cms-structure/services/cms-components.service.ts [hunk 2]
```diff
@@ -160,7 +175,8 @@ export class CmsComponentsService {
    * Return DeferLoadingStrategy for component type.
    */
   getDeferLoadingStrategy(componentType: string): DeferLoadingStrategy {
```
Comment (marlass) [inline projects/storefrontlib/src/cms-structure/services/cms-components.service.ts]: Type of the function return to update (strict mode)

File: projects/storefrontlib/src/cms-structure/services/cms-components.service.ts (hunk 1 above)
Comment (dunqan) [reply projects/storefrontlib/src/cms-structure/services/cms-components.service.ts]: 👍

File: projects/storefrontlib/src/cms-structure/services/cms-components.service.ts (hunk 2 above)
Comment (dunqan) [reply projects/storefrontlib/src/cms-structure/services/cms-components.service.ts]: 👍

**Review by marlass — APPROVED** [review]

**Pull request merged**